org.adempiere.base/src/org/compiere/db/partition/ITablePartitionService.java

```diff
@@ -99,4 +101,27 @@ public interface ITablePartitionService {
 	 * @param processInfo
 	 */
 	public void reattachPartition(MTable table, X_AD_TablePartition partition, String trxName, ProcessInfo processInfo);
+	
+	/**
+	 * Get default partition name for table
+	 * @param tableName
+	 * @return String default partition name for table
+	 */
+	public String getDefaultPartitionName(String tableName);
+	
+	/**
+	 * Parse table partition name
+	 * @param tableName
+	 * @param partitionCols
+	 * @return table partition name
+	 */
+	public String parsePartitionName(String tableName, Object[] partitionCols);
+	
+	/**
+	 * Get Partition Name of M_Product of the given level
+	 * @param tableName
+	 * @param level - 0 for the first-level partition, -1 for the lowest level
+	 * @return table partition name, or empty
+	 */
+	public String getPartitionName(Properties ctx, String tableName, int level, String trxName);
```

Comment:
The java doc is wrong, shouldn't be M_Product only.
Since only two level is supported (primary and sub), this arbitrary int level is not appropriate.